spec/integration/data_stream/fields/fields.spec.yml

```diff
@@ -534,8 +534,14 @@ spec:
           required:
             - type
         then:
-          required:
-            - object_type
+          oneOf:
+            - required:
+              - object_type
+            - properties:
+                enabled:
+                  const: false
+              required:
+                - enabled
```

Comment:
> As it is, it would still be allowed this kind of field definitions:
> 
> ```
>          - name: user_provided_metadata
>            type: object
>            object_type: keyword
>            enabled: false
> ```

We have to permit this, at least for 3.0.0-3.0.2, to avoid making a breaking change. There are some packages, at least `cloud_security_posture` using this now. I opted with the option of being more permissive.

If we consider that is problematic to allow this (it is as we have seen), I could try to add logic to disallow this starting on 3.0.3.

>    - name: user_provided_metadata
>      type: object
>      enabled: true

This mapping should not be allowed in any case and indeed it is with the current code, I will take a look and add a test for this.

Reply:
Thanks @mrodm, I think your proposal works well. I have added a JSON patch to keep things as they are before 3.0.2, so we only disallow having an `object_type` with `enabled: false` starting on 3.0.3, and we don't break current `cloud_security` posture package.

Reply:
> We have to permit this, at least for 3.0.0-3.0.2, to avoid making a breaking change. There are some packages, at least cloud_security_posture using this now. I opted with the option of being more permissive.

Ok, right we need to avoid breaking changes :+1: 

> This mapping should not be allowed in any case and indeed it is with the current code, I will take a look and add a test for this.

Thanks!